File: BACKEND/admin/services/reportService.js
const adminRepository = require("../models/adminRepository");

async function getSummary() {
  const [
    overview,
    revenueDaily,
    revenueMonthly,
    revenueBreakdown,
    appointmentBreakdown,
    topStaff,
    topServices,
    topProducts,
  ] = await Promise.all([
    adminRepository.getOverviewMetrics(),
    adminRepository.getRevenueByDay(14),
    adminRepository.getRevenueByMonth(6),
    adminRepository.getRevenueBreakdown(),
    adminRepository.getAppointmentStatusBreakdown(1),
    adminRepository.getTopStaffByRevenue(5),
    adminRepository.getTopServices(5),
    adminRepository.getTopProducts(5),
  ]);

  return {
    generatedAt: new Date().toISOString(),
    overview,
    revenue: {
      total: overview.totalRevenue,
      service: revenueBreakdown.serviceRevenue,
      product: revenueBreakdown.productRevenue,
      daily: revenueDaily,
      monthly: revenueMonthly,
    },
    appointments: {
      total: overview.totalAppointments,
      pending: overview.pendingAppointments,
      completed: overview.completedAppointments,
      cancelled: overview.cancelledAppointments,
      monthlyBreakdown: appointmentBreakdown,
      cancellationRate:
        overview.totalAppointments > 0
          ? overview.cancelledAppointments / overview.totalAppointments
          : 0,
    },
    rankings: {
      topStaff,
      topServices,
      topProducts,
    },
  };
}

module.exports = {
  getSummary,
};

File: BACKEND/admin/Services/ReportService.cs
using MyPuppy.Admin.Repositories;

namespace MyPuppy.Admin.Services;

public class ReportService
{
    private readonly AdminRepository _repository;

    public ReportService(AdminRepository repository)
    {
        _repository = repository;
    }

    public async Task<object> GetSummaryAsync()
    {
        var overviewTask = _repository.GetOverviewAsync();
        var dailyTask = _repository.GetRevenueByDayAsync(14);
        var monthlyTask = _repository.GetRevenueByMonthAsync(6);
        var breakdownTask = _repository.GetRevenueBreakdownAsync();
        var appointmentsTask = _repository.GetAppointmentStatusBreakdownAsync(1);
        var topStaffTask = _repository.GetTopStaffByRevenueAsync(5);
        var topServicesTask = _repository.GetTopServicesAsync(5);
        var topProductsTask = _repository.GetTopProductsAsync(5);

        await Task.WhenAll(
            overviewTask, dailyTask, monthlyTask, breakdownTask,
            appointmentsTask, topStaffTask, topServicesTask, topProductsTask);

        var overview = await overviewTask;
        var daily = await dailyTask;
        var monthly = await monthlyTask;
        var breakdown = await breakdownTask;
        var appointments = await appointmentsTask;
        var topStaff = await topStaffTask;
        var topServices = await topServicesTask;
        var topProducts = await topProductsTask;

        decimal cancellationRate = overview.TotalAppointments > 0
            ? (decimal)overview.CancelledAppointments / overview.TotalAppointments
            : 0m;

        return new
        {
            generatedAt = DateTime.UtcNow.ToString("o"),
            overview,
            revenue = new
            {
                total = overview.TotalRevenue,
                service = breakdown.ServiceRevenue,
                product = breakdown.ProductRevenue,
                daily,
                monthly
            },
            appointments = new
            {
                total = overview.TotalAppointments,
                pending = overview.PendingAppointments,
                completed = overview.CompletedAppointments,
                cancelled = overview.CancelledAppointments,
                monthlyBreakdown = appointments,
                cancellationRate
            },
            rankings = new
            {
                topStaff,
                topServices,
                topProducts
            }
        };
    }
}
